Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs

```diff
@@ -88,15 +99,33 @@ public async Task<string> GetIdByName(string roleName)
 
     private async Task<(string, UserProfile)> GetUserAndRoleName(UserToRoleModel model)
     {
-         var roleName = await this.roleDataService!
-            .GetByIdQuery(model.RoleId!)
-            .Select(x => x.Name!)
-            .FirstAsync();
+        var roleName = await this.roleDataService!
+           .GetByIdQuery(model.RoleId!)
+           .Select(x => x.Name!)
+           .FirstAsync();
+
+        var user = await this.usersDataService
+           .GetByIdQuery(model.UserId!)
+           .FirstAsync();
+
+        return (roleName, user);
+    }
+
+    private async Task RemoveUserFromLecturerInContestAndContestCategory(string roleName, UserProfile user)
+    {
+        if (roleName == Lecturer)
```

Comment:
Move this conditional in RolesBusinessService:85

Reply:
Done.